pkg/diagnostics/networkpod/util/log.go

```diff
@@ -0,0 +1,156 @@
+	l.Run([]string{"tc", "qdisc", "show"}, "/tc-qdisc", p)
+	l.Run([]string{"tc", "class", "show"}, "/tc-class", p)
+	l.Run([]string{"tc", "filter", "show"}, "/tc-filter", p)
+	l.Run([]string{"systemctl", "cat", "docker.service"}, "/docker-unit-file", p)
+	l.Run([]string{"cat", "$(systemctl cat docker.service | grep EnvironmentFile.*openshift-sdn | awk -F=- '{print $2}')"}, "/docker-network-file", p)
+}
+
+func (l *LogInterface) LogMaster(logdir string) {
+	p := logdir
+
+	l.LogSystem(p)
+	l.LogServices(p)
+
+	l.Run([]string{"oc", "get", "nodes", "-o", "yaml"}, "/nodes", p)
+	l.Run([]string{"oc", "get", "pods", "--all-namespaces", "-o", "yaml"}, "/pods", p)
+	l.Run([]string{"oc", "get", "services", "--all-namespaces", "-o", "yaml"}, "/services", p)
+	l.Run([]string{"oc", "get", "endpoints", "--all-namespaces", "-o", "yaml"}, "/endpoints", p)
+	l.Run([]string{"oc", "get", "routes", "--all-namespaces", "-o", "yaml"}, "/routes", p)
+	l.Run([]string{"oc", "get", "clusternetwork", "-o", "yaml"}, "/clusternetwork", p)
+	l.Run([]string{"oc", "get", "hostsubnets", "-o", "yaml"}, "/hostsubnets", p)
+	l.Run([]string{"oc", "get", "netnamespaces", "-o", "yaml"}, "/netnamespaces", p)
+}
+
+func (l *LogInterface) LogServices(logdir string) {
+	p := logdir
+
+	type Service struct {
+		name   string
+		suffix string
+		args   string
+	}
+	allServices := []Service{
+		{suffix: "master", args: "master"},
+		{suffix: "master-controllers", args: "master controllers"},
+		{suffix: "api", args: "master api"},
+		{suffix: "node", args: "node"},
+	}
+	foundServices := []Service{}
+
+	for _, sysDir := range []string{"/etc/systemd/system", "/usr/lib/systemd/system"} {
+		for _, prefix := range []string{"openshift", "origin", "atomic-openshift"} {
+			for _, service := range allServices {
+				service.name = fmt.Sprintf("%s-%s", prefix, service.suffix)
+				servicePath := fmt.Sprintf("%s/%s.service", sysDir, service.name)
+				if _, err := os.Stat(servicePath); err == nil {
+					foundServices = append(foundServices, service)
+				}
+			}
+		}
+	}
+
+	for _, service := range foundServices {
+		l.Run([]string{"journalctl", "-u", service.name, "-r", "-n", "5000"}, fmt.Sprintf("/journal-%s", service.name), p)
+		l.Run([]string{"systemctl", "show", service.name}, fmt.Sprintf("/systemctl-show-%s", service.name), p)
+
+		configFile := l.getConfigFileForService(service.name, service.args)
+		if len(configFile) > 0 {
+			l.Run([]string{"cat", configFile}, fmt.Sprintf("/config-%s", service.name), p)
+		}
+	}
+}
+
+func (l *LogInterface) LogSystem(logdir string) {
+	p := logdir
+
+	l.Run([]string{"journalctl", "--boot", "-r", "-n", "5000"}, "/journal-boot", p)
+	l.Run([]string{"nmcli", "--nocheck", "-f", "all", "dev", "show"}, "/nmcli-dev", p)
+	l.Run([]string{"nmcli", "--nocheck", "-f", "all", "con", "show"}, "/nmcli-con", p)
+	l.Run([]string{"head", "-1000", "/etc/sysconfig/network-scripts/ifcfg-*"}, "/ifcfg", p)
```

Comment:
The use of "head -1000" in debug.sh was a hack: "1000" basically means "infinity" here, and we're just using "head" rather than "cat" because "head" prefixes each file with its filename when operating on multiple files:

    ==> /etc/sysconfig/network-scripts/ifcfg-enp0s25 <==
    TYPE=Ethernet
    BOOTPROTO=dhcp
    ...
    
    ==> /etc/sysconfig/network-scripts/ifcfg-lo <==
    DEVICE=lo
    IPADDR=127.0.0.1
    ...

as with the use of `$(...)` above, it feels wrong-ish to do this if we're not a hacky shell script any more...